router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/marketobulkupload.go

```diff
@@ -203,3 +213,8 @@ func (b *MarketoBulkUploader) Upload(destination *backendconfig.DestinationT, as
 	return uploadResponse
 
 }
+
+func (b *MarketoBulkUploader) RetrieveImportantKeys(metadata map[string]interface{}, retrieveKeys string) ([]int64, error) {
```

Comment:
can metadata be a struct?

Reply:
address in the pr https://github.com/rudderlabs/rudder-server/pull/3483